File: src/ios/App/Models/Order.m
// ==============================================================================
//
// This file is part of the WelHair
//
// Create by Welfony <support@welfony.com>
// Copyright (c) 2013-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================
#import "Order.h"
#import "OpitionItem.h"
@implementation Order

- (id)init
{
    self = [super init];
    if(self){
        self.productSelectedSpecs = [NSMutableDictionary dictionary];
        self.count = 1;
        self.status = OrderStatusEnum_UnPaid;
    }
    return self;
}
// check if need pick spec
- (NSString *)unSelectedSpecStr
{
    NSMutableString *str = [NSMutableString string];
    for (NSDictionary *dic  in self.product.specList) {
        int specId = [[dic objectForKey:@"SpecId"] intValue];
        if(![self.productSelectedSpecs objectForKey:@(specId)]){
            [str appendFormat:@"%@,",[dic objectForKey:@"Title"]];
            continue;
        }
    }
    if(str.length > 0){
        [str deleteCharactersInRange:NSMakeRange(str.length-1, 1)];
    }
    return str;
}

- (NSString *)selectedSpecStr
{
    NSMutableString *str = [NSMutableString string];
    for (OpitionItem *item in self.productSelectedSpecs.allValues) {
        [str appendFormat:@"%@,",item.title];
    }
    if(str.length > 0){
        [str deleteCharactersInRange:NSMakeRange(str.length-1, 1)];
    }
    return str;

}
@end

File: src/www/api/routes/order.php
<?php

// ==============================================================================
//
// This file is part of the WelStory.
//
// Create by Welfony Support <support@welfony.com>
// Copyright (c) 2012-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================

use Welfony\Controller\API\OrderController;

$app->get('/users/:userId/orders', function ($userId) use ($app) {
    $controller = new OrderController();
    $controller->listAllOrdersByUserId($userId);
})->conditions(array('userId' => '\d{1,10}'));

$app->post('/orders', function () use ($app) {
    $ctrl = new OrderController();
    $ctrl->create();
});

$app->post('/orders/:orderId/pay', function ($orderId) use ($app) {
    $ctrl = new OrderController();
    $ctrl->pay($orderId);
})->conditions(array('orderId' => '\d{1,10}'));